### DISCUSSION TITLE
How to plot output labels from model.val()

### DISCUSSION THREAD

**Discussion by hamedgorji** [ultralytics/ultralytics#6497] [General]
Hello everyone,

I utilized the command metrics = model.val(data="Data/data-test.yaml", split='test', save_txt=True) and it created a "labels" folder. Is there an existing command to plot and save all the labels along with their corresponding images, or do I need to write custom code for this?

Thanks

**Comment by pderrenger** [comment]
@hamedgorji hello! 😊

After running `model.val()` with `save_txt=True`, the labels are saved in a "labels" folder in a YOLO format, which is great for further analysis but not directly for visualization.

To visualize and save the labels on the corresponding images, you can use the `Predict` mode with the `save=True` argument. This will run inference on your dataset and save the images with the predictions drawn on them. Here's an example of how you can do this:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/your/model.pt')

# Perform prediction on your test dataset and save the images
results = model.predict(source='path/to/your/test/images', save=True)
```

This will save the images with the predictions in the `runs/detect/exp` directory by default. You can also specify the `project` and `name` parameters to save the results in a custom directory.

I hope this helps! If you have any more questions, feel free to ask. Happy coding! 🚀
